### PR TITLE
[sc-93181] Create circle job to tag and publish new release

### PR DESCRIPTION
**Pull request by jess-ingraham**
This PR adds a circle flow that runs on the `main` branch. I used fastlane since it has some built in functionality with github that was really useful. This workflow will only run on the main branch. This is slightly different from the android repo because there's no version stored in the codebase, we can just use the tag from the previous release.

The steps are as follows
1. See new commit from design-token repo pushed to `main`
2. CircleCI picks up the new commit
3. Get the most recent tag from github and use it to update the minor version for the next release
4. Creates a new tag with the new version and publishes a release with the new version as the title and the lastest commit message in the notes (this will be the message from the design-tokens repo)

If we want to update the major version whenever it's needed (like when this library is fully ready for use) we can do that manually and just skip the circle workflow using `[ci skip]` in the commit message

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -18,8 +18,5 @@ let package = Package(
         .target(
             name: "iosDesignTokens",
             dependencies: []),
-//        .testTarget(
-//            name: "iosDesignTokensTests",
-//            dependencies: ["iosDesignTokens"]),
```
Comment (HeinousGames) [inline Package.swift:27]: do we still need these comments?

Comment (jess-ingraham) [reply Package.swift:27]: this change is removing the comments

**Review by HeinousGames — APPROVED** [review]

File: Package.swift (hunk 1 above)
Comment (HeinousGames) [reply Package.swift:27]: whoops!

**Review by flitterio — APPROVED** [review]

File: fastlane/Fastfile [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    # if the major version needs to change, that needs to happen manually
+    parts = last_tag.split(".")
+    parts[1] = parts[1].to_i + 1
+    parts[2] = "0"
+    new_tag = parts.join(".")
+    
+    add_git_tag(tag: new_tag)
+    push_git_tags
+    
+    set_github_release(
```
Comment (zstecker) [inline fastlane/Fastfile:39]: neat!

**Review by zstecker — APPROVED** [review]
looks good!

**Pull request merged**